numpy/polynomial/polyutils.py

```diff
@@ -542,17 +542,12 @@ def _valnd(val_f, c, *args):
     c, args :
         See the ``<type>val<n>d`` functions for more detail
     """
-    try:
-        args = tuple(np.array(args, copy=False))
-    except Exception:
-        # preserve the old error message
-        if len(args) == 2:
+    if len(args) > 2:
+        if len(args[0]) != len(args[1]) or len(args[1]) != len(args[2]):
             raise ValueError('x, y, z are incompatible')
-        elif len(args) == 3:
+    elif len(args) > 1:
+        if len(args[0]) != len(args[1]):
             raise ValueError('x, y are incompatible')
-        else:
-            raise ValueError('ordinates are incompatible')
```

Comment:
This error is no longer raised when called with 4 arrays that do not match

Reply:
Why would we need it? As far as I can tell `_valnd` is always called with 2 or three values for `args`.

Reply:
All of these ND functions were intended to be truly ND generalizations waiting only for a public API to be designed. It seems foolish to limit it to 2 and 3d for the sake of error checking.

Also, this changes the behavior on 2d arrays. Checking `.shape` after calling asanyarray on each would do the job.

Reply:
Actually, probably best would be:
```
try:
    args = np.broadcast_arrays(args)
except ValueError:
    raise ValueError ("specific message")

Reply:
`args = np.broadcast_arrays(args)` doesn't work, it broadcasts the args instead of simply checking that they match, which causes problems when actually calling the function. I pushed a change with `asanyarray` and checking that the shapes are equal. That is more strict than `broadcast_arrays`, are the arguments meant to be broadcastable?

Reply:
Not sure if they're meant to be broadcastable, but they certainly weren't before - what you have looks fine, thank!